src/main/webapp/app/exercises/file-upload/manage/file-upload-exercise-update.component.ts

```diff
@@ -66,10 +66,24 @@ export class FileUploadExerciseUpdateComponent implements OnInit {
     }
 
     /**
-     * Returns to previous state, which should be always the page of selected course
+     * Revert to the previous state, equivalent with pressing the back button on your browser
+     * Returns to the detail page if there is no previous state and we edited an existing exercise
+     * Returns to the overview page if there is no previous state and we created a new exercise
+     * Returns to the exercise group page if we are in exam mode
      */
     previousState() {
-        window.history.back();
+        if (window.history.length > 1) {
+            window.history.back();
+        } else if (this.isExamMode) {
+            // If we're editing we need to go an extra step back since there is no detail page for exercise groups
+            if (this.fileUploadExercise.id) {
+                this.router.navigate(['../../../../'], { relativeTo: this.activatedRoute });
```

Comment:
Hmm is it possible to enter the name of the component where you want to navigate to instead of using these paths? This can cause problems if ever (ever) we change what pages are shown in the future.

Reply:
I agree